### PR TITLE
#212 - HeadProxySlice fixed

### LINKED ISSUE #212: HeadProxySlice issues
There are two problems with `HeadProxySlice`:
1. maven-central returns 421 on requests with `Host` header and vertx adds such header, so we should not proxy headers in this slice
2. response headers (headers from proxy, from maven central for example) are not returned in response

Let's fix this and add corresponding tests.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part #212 
I've fixed `HeadProxySlice` do not pass headers on request to proxy and return obtained headers from proxy to the client.

### CODE REVIEW

File: src/test/java/com/artipie/maven/http/MavenProxySliceITCase.java [hunk 1]
```diff
@@ -90,15 +80,17 @@
...
-                new StorageCache(this.storage)
+        this.server = new VertxSliceServer(
+            MavenProxySliceITCase.VERTX,
+            new LoggingSlice(
+                new MavenProxySlice(
+                    this.client,
+                    URI.create("https://repo.maven.apache.org/maven2"),
+                    Authenticator.ANONYMOUS,
+                    new StorageCache(this.storage)
+                )
```
Comment (olegmoz) [inline src/test/java/com/artipie/maven/http/MavenProxySliceITCase.java:91]: @olenagerasimova could you please explain why it is needed to wrap `Slice` into real HTTP server here? To me it seems easier to test `Slice` itself without additional level of abstraction

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one question

**Review by olegmoz — APPROVED** [review]
@olenagerasimova as we discussed, expecting you to write a unit test for this issue as well

**Comment by 0crat** [comment]
Job `gh:artipie/maven-adapter#213` is not assigned, can't get performer

**Pull request merged**